### PR TITLE
Fft makeover

### PR DESCRIPTION
**Pull request by Jellevanderwerff**
In order:
* power is now returned (ff44117)
* dc offset removal has been added as an argument that defaults to True (offset removed; 41cbc1f)
* Nyquist was already implemented of course
* I don't see a difference between the two options actually.

### CODE REVIEW

**Comment by Jellevanderwerff** [comment]
Can be merged (despite test failing, which is to do with an unrelated issue)

**Comment by YannickJadoul** [comment]
Huh, sorry, I pulled this and rebased, but something must have gone wrong. Figuring out what went wrong!

**Comment by YannickJadoul** [comment]
Ah, I think I didn't fetch before :-|
Cleaning upç

File: thebeat/stats.py [hunk 1]
```diff
@@ -609,13 +610,25 @@ def fft_values(
     The number of cycles per unit can be interpreted as the beat frequency. For instance, 2 cycles for a unit size of
     1000 ms means a beat frequency of 2 Hz.
 
+    Note
+    ----
+    If a sequence ends with an event, the last event is not included in the calculation of the Fourier transform.
+    This is because the Fourier transform assumes that the sequence can be repeated indefinitely.
+
```
Comment (YannickJadoul) [inline thebeat/stats.py:617]: You could also phrase this in a way that it's about the intervals, not the events. Cause that's basically what it is: A potential event at the end of the sequence will be directly followed by the first event anyway.

Comment (YannickJadoul) [reply thebeat/stats.py:617]: Oh, but more importantly! What happens for a sequence where the first event isn't at time 0? Is it padded with 0s before, or does it just start at the first events? How does this interact with this last event case?
Can we make this explicit in the docs and add a short test?

File: thebeat/stats.py [hunk 2]
```diff
@@ -637,17 +650,23 @@ def fft_values(
 
     # Make a sequence of ones and zeroes
     timeseries = thebeat.helpers.sequence_to_binary(sequence, resolution=step_size)
-    duration = sequence.duration
+    timeseries = timeseries if sequence.end_with_interval is True else timeseries[:-1]
+
+    # Remove DC component if necessary
+    if remove_dc:
```
Comment (YannickJadoul) [inline thebeat/stats.py:656]: Can we get a test that this makes a difference? And also test (possibly in the same test) whether the DC component is indeed equal to 0, then?

File: thebeat/stats.py [hunk 3]
```diff
@@ -637,17 +650,23 @@ def fft_values(
 
     # Make a sequence of ones and zeroes
     timeseries = thebeat.helpers.sequence_to_binary(sequence, resolution=step_size)
-    duration = sequence.duration
+    timeseries = timeseries if sequence.end_with_interval is True else timeseries[:-1]
```
Comment (YannickJadoul) [inline thebeat/stats.py:653]: Is `helpers` public, or just internal?
If it's public and documented, maybe this "include final event" could be an optional keyword argument to `sequence_to_binary`? I can imagine this to be nice bonus if people want to use this?

**Review by YannickJadoul — APPROVED** [review]
LGTM!
Couple of suggetsions, but nothing crucial apart from the "first even not at time 0" note

File: thebeat/stats.py (hunk 1 above)
Comment (Jellevanderwerff) [reply thebeat/stats.py:617]: The latter is taken into account in the ``sequence_to_binary`` function

Comment (YannickJadoul) [reply thebeat/stats.py]: Fixed by later commits, great!

File: thebeat/stats.py (hunk 3 above)
Comment (YannickJadoul) [reply thebeat/stats.py]: The way it was changed wrt. the time of the first onset, this isn't relevant anymore.

**Pull request merged**